src/parser/GenericParser.test.ts

```diff
@@ -0,0 +1,281 @@
+        const expectedFileMetricsMap = new Map([
+            [FileMetric.linesOfCode, { metricName: FileMetric.linesOfCode, metricValue: 5 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindTwoFiles();
+        });
+
+        const treeParserSpied = jest
+            .spyOn(TreeParser, "parse")
+            .mockImplementation((file, assumedLanguage) => {
+                return Promise.resolve({
+                    fileExtension: file.fileExtension,
+                    filePath: file.filePath,
+                    language: Language.CPlusPlus,
+                    tree: tree,
+                });
+            });
+
+        const calculateMetricsSpied = jest
+            .spyOn(MetricCalculator.prototype, "calculateMetrics")
+            .mockImplementation(async (parsedFilePromise) => [
+                (await parsedFilePromise).filePath,
+                expectedFileMetricsMap,
+            ]);
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid"));
+        const actualResult = await parser.calculateMetrics();
+
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([
+                ["clearly/invalid/path1.cc", expectedFileMetricsMap],
+                ["clearly/invalid/path2.cpp", expectedFileMetricsMap],
+            ])
+        );
+        expect(calculateMetricsSpied).toHaveBeenCalledTimes(2);
+        expect(treeParserSpied).toHaveBeenCalledTimes(2);
+    });
+
+    it("should return the appropriate value and call the appropriate functions in case there is an error while parsing the tree", async () => {
+        const expectedFileMetricsMap = new Map([
+            ["ERROR", { metricName: "ERROR", metricValue: -1 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindFilesAsync();
+        });
+
+        jest.spyOn(TreeParser, "parse").mockImplementation(async (file, assumedLanguage) => {
+            throw new Error("Baaaaaah");
+        });
+
+        jest.spyOn(MetricCalculator.prototype, "calculateMetrics").mockResolvedValue([
+            "clearly/invalid/path.cpp",
+            expectedFileMetricsMap,
+        ]);
+
+        const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {
+            /* Do nothing */
+        });
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid/path.cpp"));
+        const actualResult = await parser.calculateMetrics();
+
+        // TODO: do no longer include errors into the list of results, use "info" field instead.
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([["clearly/invalid/path.cpp", expectedFileMetricsMap]])
+        );
```

Comment:
I like to have // given // when // then comment within the tests to have a clearer structure. 

```suggestion
        // given
        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
        findFilesAsyncMock.mockImplementation((config) => {
            return mockedFindFilesAsync();
        });

        jest.spyOn(TreeParser, "parse").mockImplementation(async (file, assumedLanguage) => {
            throw new Error("Baaaaaah");
        });

        jest.spyOn(MetricCalculator.prototype, "calculateMetrics").mockResolvedValue([
            "clearly/invalid/path.cpp",
            expectedFileMetricsMap,
        ]);

        const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {
            /* Do nothing */
        });

        const parser = new GenericParser(getParserConfiguration("clearly/invalid/path.cpp"));
        
        // when
        const actualResult = await parser.calculateMetrics();

        // then
       const expectedFileMetricsMap = new Map([
            ["ERROR", { metricName: "ERROR", metricValue: -1 }],
        ]);
        expect(actualResult.fileMetrics).toEqual(
            new Map([["clearly/invalid/path.cpp", expectedFileMetricsMap]])
        );
```

The given block contains your setup and mocks
The when block contains the function or action that is executed
Then contains assertions.